include/sys/vdev_impl.h

```diff
@@ -272,6 +272,8 @@ struct vdev {
 	    MAX_UBERBLOCK_SHIFT)
 #define	VDEV_UBERBLOCK_COUNT(vd)	\
 	(VDEV_UBERBLOCK_RING >> VDEV_UBERBLOCK_SHIFT(vd))
+#define	VDEV_UBERBLOCK_SLOT(vd, txg)	\
+	(txg % VDEV_UBERBLOCK_COUNT(vd))
```

Comment:
Minor note - this will be expensive on 32-bit systems where txg is a 64-bit variable.  You might want to consider `(unsigned long)txg % VDEV_UBERBLOCK_COUNT(vd)`.  That has some discontinuity every 4B TXGs, but at least for 119-127 uberblocks the modulus `0xffffffff % uberblock_count` is no less than 3 so when it wraps we will still have at least the last 3 uberblocks.

That said, ZFS on 32-bit systems likely already has a large number of such issues, so one more probably isn't fatal.